src/stream/src/executor/merge.rs

```diff
@@ -147,18 +147,26 @@ impl MergeExecutor {
                     if let Some(update) =
                         barrier.as_update_merge(self.actor_context.id, self.upstream_fragment_id)
                     {
-                        assert!(update.new_upstream_fragment_id.is_none());
+                        let new_upstream_fragment_id = update
+                            .new_upstream_fragment_id
+                            .unwrap_or(self.upstream_fragment_id);
+                        let added_upstream_actor_id = update.added_upstream_actor_id.clone();
+                        let removed_upstream_actor_id: HashSet<_> =
+                            if update.new_upstream_fragment_id.is_some() {
+                                select_all.upstream_actor_ids().iter().copied().collect()
```

Comment:
Remove all actors if fragment id is changed.

https://github.com/risingwavelabs/risingwave/blob/54a92ddd2befe51ce73905aaa2a721fca0b20f74/proto/stream_plan.proto#L47-L50